### PR TITLE
FINERACT-822 enable UnnecessaryDefaultInEnumSwitch error

### PR DESCRIPTION
**Pull request by percyashu**
FINERACT-822

### CODE REVIEW

File: fineract-provider/src/main/java/org/apache/fineract/adhocquery/service/AdHocScheduledJobRunnerServiceImpl.java
Comment (vorburger) [inline fineract-provider/src/main/java/org/apache/fineract/adhocquery/service/AdHocScheduledJobRunnerServiceImpl.java]: wait... wat ??  By doing this, it will ALWAYS throw, no? This will completely break. @awasum please help..

**Review by vorburger — CHANGES_REQUESTED** [review]

File: fineract-provider/src/main/java/org/apache/fineract/adhocquery/service/AdHocScheduledJobRunnerServiceImpl.java
Comment (awasum) [reply fineract-provider/src/main/java/org/apache/fineract/adhocquery/service/AdHocScheduledJobRunnerServiceImpl.java]: @percyashu ..Maybe remove this exception? and put it back in the default case inside the } on line 91 above?

Also if you want to add a default step, you need to check the `ReportRunFrequency.fromId(adhoc.getReportRunFrequency()` method from line 70 above and see how many different cases needs to be treated, then try to handle them all. Do we absolutely need a default case in all implementations of Switch statement?
As seen here: https://errorprone.info/bugpattern/UnnecessaryDefaultInEnumSwitch, dont put a default statement if all the cases have been handled. Maybe skip it? Will that throw an error?

Or given that ReportRunFrequency is an enum and we/Fineract is already handling all the 5 cases here, we can just throw a `new AdHocScheduledJobFrequencyException()` error here as default case with a message like "Correct Frequency not specified, Check the API docs"... @vorburger  @ptuomola your take?

**Review by awasum — CHANGES_REQUESTED** [review]
Address feedback...Please.

**Comment by awasum** [comment]
Fix conflicts too.

**Comment by percyashu** [comment]
@awasum I will do

**Comment by vorburger** [comment]
@ptuomola was wondering if you would like to help code-review this PR as well? Just because it's this is very similar & seems related to #1006 on which you have engaged. If interested, then I or @awasum would merge this only after your LGTM.

File: fineract-provider/src/main/java/org/apache/fineract/adhocquery/service/AdHocScheduledJobRunnerServiceImpl.java
Comment (ptuomola) [reply fineract-provider/src/main/java/org/apache/fineract/adhocquery/service/AdHocScheduledJobRunnerServiceImpl.java]: Isn't this ErrorProne check specifically highlighting switch statements where the default case is never used, because we are already handling all the cases specifically? Given that, wouldn't the right solution just be to remove the default including the throw, and not add anything?

File: fineract-provider/src/main/java/org/apache/fineract/infrastructure/documentmanagement/contentrepository/ContentRepositoryUtils.java
Comment (ptuomola) [inline fineract-provider/src/main/java/org/apache/fineract/infrastructure/documentmanagement/contentrepository/ContentRepositoryUtils.java]: Same here - shouldn't we just remove the default case altogether, not add anything back?

File: fineract-provider/build.gradle
Comment (vorburger) [inline fineract-provider/build.gradle]: Is this being removed (not moved) intentional?

Comment (percyashu) [reply fineract-provider/build.gradle]: It was moved to the error list in a previous PR.  Don't know how it is here (maybe during merge) so I just removed it

File: fineract-provider/src/main/java/org/apache/fineract/adhocquery/service/AdHocScheduledJobRunnerServiceImpl.java
Comment (percyashu) [reply fineract-provider/src/main/java/org/apache/fineract/adhocquery/service/AdHocScheduledJobRunnerServiceImpl.java]: @ptuomola I thought so to but it requires a return. 
@vorburger I thought if the case is not found, switch default is used with exception thrown and if case is not found and no default then switch is not executed then exception is thrown. That's why I did it like this. Just undid what I did and added a suppressedwarning

**Comment by percyashu** [comment]
@ptuomola @vorburger @awasum  LGTY?

File: fineract-provider/src/main/java/org/apache/fineract/adhocquery/service/AdHocScheduledJobRunnerServiceImpl.java
Comment (awasum) [inline fineract-provider/src/main/java/org/apache/fineract/adhocquery/service/AdHocScheduledJobRunnerServiceImpl.java]: If you have suppressed this check..Do we still need this default?

**Comment by awasum** [comment]
@percyashu  Address last set of feedback here and maybe @ptuomola will give it a LGTH

**Review by awasum — APPROVED** [review]

File: fineract-provider/src/main/java/org/apache/fineract/adhocquery/service/AdHocScheduledJobRunnerServiceImpl.java
Comment (percyashu) [reply fineract-provider/src/main/java/org/apache/fineract/adhocquery/service/AdHocScheduledJobRunnerServiceImpl.java]: Was my mistake wasn't not suppose to change the exception will revert it. Yeah we need it because Error prone thinks it is not necessary since all cases are specified.

**Comment by vorburger** [comment]
@ptuomola does this one have LGTM from you now?

**Review by ptuomola — APPROVED** [review]
Looks good to me!

**Pull request merged**
